core/src/test/scala/org/apache/spark/deploy/SparkHadoopUtilSuite.scala

```diff
@@ -110,16 +110,18 @@ class SparkHadoopUtilSuite extends SparkFunSuite {
   test("SPARK-40640: aws credentials from environment variables") {
     val hadoopConf = new Configuration(false)
     SparkHadoopUtil.appendS3CredentialsFromEnvironment(hadoopConf,
-      "access-key", "secret-key", "session-token")
+      "endpoint", "access-key", "secret-key", "session-token")
     val source = "Set by Spark on " + InetAddress.getLocalHost + " from "
+    assertConfigMatches(hadoopConf, "fs.s3a.endpoint", "endpoint", source)
     assertConfigMatches(hadoopConf, "fs.s3a.access.key", "access-key", source)
     assertConfigMatches(hadoopConf, "fs.s3a.secret.key", "secret-key", source)
     assertConfigMatches(hadoopConf, "fs.s3a.session.token", "session-token", source)
   }
 
   test("SPARK-19739: S3 session token propagation requires access and secret keys") {
     val hadoopConf = new Configuration(false)
-    SparkHadoopUtil.appendS3CredentialsFromEnvironment(hadoopConf, null, null, "session-token")
+    SparkHadoopUtil.appendS3CredentialsFromEnvironment(
+      hadoopConf, null, null, null, "session-token")
     assertConfigValue(hadoopConf, "fs.s3a.session.token", null)
```

Comment:
nit:
```suggestion
    SparkHadoopUtil.appendS3CredentialsFromEnvironment(
      hadoopConf, null, null, null, "session-token")
    assertConfigValue(hadoopConf, "fs.s3a.endpoint", null)  
    assertConfigValue(hadoopConf, "fs.s3a.session.token", null)
```

Reply:
Oh.. Got it.

Reply:
How about this? This test the default value and the requirements of access/secret keys, @viirya ?
```scala
test("SPARK-45404: aws endpoint propagation requires access and secret keys") {
   val hadoopConf = new Configuration(false)
   SparkHadoopUtil.appendS3CredentialsFromEnvironment(
     hadoopConf, "endpoint", null, null, null)
   assertConfigValue(hadoopConf, "fs.s3a.endpoint", null)
}
```

Reply:
I think it tests the default value. Just wonder why it tests the requirements of access/secret keys (as they are both nulls).

Reply:
Oh, got it. They (keyId and accessKey) must be set in order to set others.

Reply:
Looks good to me.